yew-macro/src/html_tree/html_tag/mod.rs

```diff
@@ -132,7 +132,7 @@ impl ToTokens for HtmlTag {
         });
         let set_classes = classes.iter().map(|classes_form| match classes_form {
             ClassesForm::Tuple(classes) => quote! {
-                #vtag.add_classes(vec![#(&(#classes)),*]);
+                #vtag.set_classes(<::yew::virtual_dom::Classes as ::std::convert::From::<Vec<&str>>>::from(vec![#(&(#classes)),*]));
```

Comment:
Is it possible to do `::yew::virtual_dom::Classes::from(vec![#(&(#classes)),*])`?

Reply:
It is not possible.
When writing `<div class=("class-1", "class-2".to_string()) />` the vector is resulting in `vec![&("class-1"), &("class-2".to_string())]`, the first type will be `&&str`, and the second one will be `&String` and then the compiler says that there are mismatched types, but if there is a type annotation, the compiler can tread the vector as Vec<&str> (via a Deref coercion).

Reply:
What about changing the Tuple form into this `::yew::virtual_dom::Classes::default()#(.extend(#classes))*`.
This would allow each tuple member to be of a different type, as long as `std::convert::Into<::yew::virtual_dom::Classes>` is implemented for each tuple member.
E.g. the following would be possible:
```rust
html! { <div class=("class-1", String::from("class-2"), vec!["class-3", "class-4"])></div> }
```
And if `impl<T: AsRef<str>> From<Option<T>> for Classes` would be implemented, the following would also be possible:
```rust
html! { <div class=("class-1", Some("class-2"), "class-3")></div> }
```
This would solve #936 without using the workaround.

Reply:
Yeah, that sounds awesome!